### PR TITLE
20260416 bond coverage [GEN-7371]

### PR DESCRIPTION
**Pull request by kronael**
Subtract minBondEpoch from bondGoodForNEpochs so that it has a better interpretation.

Now its 0 if and only if you are getting changed and undelegated.

### CODE REVIEW

File: packages/ds-sam-sdk/src/constraints.ts [hunk 1]
```diff
@@ -245,7 +245,8 @@ export class AuctionConstraints {
       bondBalanceSol - (revShare.onchainDistributedPmpe / 1000) * protectedStakeSol,
     )
     validator.bondGoodForNEpochs =
```
Comment (ochaloup) [inline packages/ds-sam-sdk/src/constraints.ts:248]: Can we maybe add a short comment (again :-)) explaining why this is now the calculation?

File: packages/ds-sam-sdk/src/constraints.ts [hunk 2]
```diff
@@ -245,7 +245,8 @@ export class AuctionConstraints {
...
-      bondBalanceForBids / ((revShare.expectedMaxEffBidPmpe / 1000) * validator.marinadeActivatedStakeSol)
+      bondBalanceForBids / ((revShare.expectedMaxEffBidPmpe / 1000) * validator.marinadeActivatedStakeSol) -
```
Comment (ochaloup) [inline packages/ds-sam-sdk/src/constraints.ts:249]: We consider this behavior correct - when `marinadeActivatedStakeSol` is `0`, the division produces `Infinity` and the result becomes `null` in JSON, I think.
Maybe we could add this edge case to a test to make sure this behavior is intentional.

**Review by ochaloup — DISMISSED** [review]
Maybe to update the `ARCHITECTURE.md` to be aligned with the change of the calculation.

File: packages/ds-sam-sdk/src/constraints.ts (hunk 2 above)
Comment (kronael) [reply packages/ds-sam-sdk/src/constraints.ts:249]: I will add these.  We should maybe handle those even better in a subtle way...

Comment (kronael) [reply packages/ds-sam-sdk/src/constraints.ts:249]: Or is it better to cap it at 1k ?  Might be better for the consumers as NaN might mean something different than Inf.

**Comment by kronael** [comment]
Thank you for the comments.  Updated and added one more test which revelealed more bugs, fixed them.

**Review by ochaloup — DISMISSED** [review]

**Review by ochaloup — APPROVED** [review]
